src/server/pmix_server_get.c

```diff
@@ -490,6 +490,7 @@ static pmix_status_t create_local_tracker(char nspace[], pmix_rank_t rank,
     if (NULL == lcd){
         return PMIX_ERR_NOMEM;
     }
+    PMIX_RETAIN(cbdata);
```

Comment:
Hold on a second - you cannot do a retain on cbdata as it is a void*, which means you cannot know for certain that it is a PMIX_OBJECT

Reply:
I'm not seeing the code path that is causing a problem. Each peer that requests some info has its own object added to the tracker, and that object has a pointer to the cbdata provided at the time of request. There is nobody releasing it unless the request fails to be accepted.

Can you explain what problem you are actually seeing?

Reply:
The problem is after the first request is added to loc_reqs,  for the following requests' cbdata (cd at pmix_server.c:3071) has the same memory address as the first request's cd , then the loc_reqs context will be changed.  

I will dig more to see why this happens.

Reply:
Yeah, that doesn't sound right at all. We do a PMIX_NEW each time, so hard to see how it comes back with the same address.

Reply:
I think the memory is released somehow and then get reused, i am debugging to see when and where it is released.

Reply:
it will be released when the callback function gets executed and we release the overall object

Reply:
I think it is still in local_reqs list when it is released.

Reply:
?? Then it would assert, yes? Are you sure you are at the top of master?

Reply:
For the first request if it is not satisfied and then create a local_tracker, this request is added in loc_reqs, if the return value of pmix_server_get (pmix_server.c:3072 ) is not PMIX_SUCCESS, can we do a PMIX_RELEASE(cd) below?

Reply:
Why don't you bring this to the meeting on Thurs so we can discuss it? I'm having trouble following the logic of the problem - I still don't understand how a memory address is being free'd and then reused.

Reply:
Sure